src/Microsoft.Health.Dicom.Core/Configs/RetrieveConfiguration.cs

```diff
@@ -17,4 +20,21 @@ public class RetrieveConfiguration
     /// This is the size of the buffer
     /// </summary>
     public int LazyResponseStreamBufferSize { get; } = 1024 * 1024 * 4; //4 MB
+
+    /// <summary>
+    /// Gets or sets the maximum number of tasks that should be concurrently scheduled to read for a single metadata request.
+    /// </summary>
+    /// <value>A positive number or <c>-1</c> for unbounded parallelism.</value>
+    [Range(1, int.MaxValue)]
+    public int MaxDegreeOfParallelism { get; set; } = Environment.ProcessorCount * 4;
```

Comment:
Why the CPU count * 4 here?

Reply:
I've also changed to be `-1` as that will default to the `TaskScheduler`'s maximum concurrency (which is by default `int.MaxValue`. If we asked for something higher, it would be capped to the scheduler's maximum by the `Parallel` class